source/MaterialXCore/Definition.cpp

```diff
@@ -86,8 +86,7 @@ InterfaceElementPtr NodeDef::getImplementation(const string& target, const strin
     for (InterfaceElementPtr interface : interfaces)
     {
         if (interface->isA<Implementation>() ||
-            !targetStringsMatch(interface->getTarget(), target) ||
-            !isVersionCompatible(interface))
```

Comment:
To make it so no version is required on a nodegraph to provide a nodedef<->nodegraph match. Removing this
check.